src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java

```diff
@@ -47,12 +48,39 @@ public SecurityGroupWithIdAndRole createSecurityGroup(SecurityGroup securityGrou
         return securityGroupWithIdAndRole;
     }
 
-    public List<SecurityGroupWithIdAndRole> getSecurityGroups() {
+    public List<SecurityGroupWithIdAndRole> getSecurityGroups(List<Integer> roleIds, Integer courthouseId) {
         List<SecurityGroupEntity> securityGroupEntities = securityGroupRepository.findAll();
```

Comment:
I'm wondering if we should be trying to do a custom finder here on the `SecurityGroupRepository`?
Probably one to confirm with @hemantasharma1129 / @chrisbellingham-hmcts

Reply:
The approach we've taken elsewhere is to construct the DB query dynamically, based on the provided filter params (e.g `UserSearchQueryImpl.java`). I think this approach is a good one for large datasets, albeit at the cost of some complexity.

However since the `security_group` table will be relatievly small (within the order of hundreds of records), the findAll approach is probably good enough. There's a benefit to readability in doing the filtering after-the-fact with the java streams api. Let's revisit if performance testing tells a different story though.